src/Compilers/CSharp/Portable/Symbols/BaseTypeAnalysis.cs

```diff
@@ -93,8 +93,7 @@ private static void StructDependsClosure(NamedTypeSymbol type, HashSet<Symbol> p
         /// IsManagedType is simple for most named types:
         ///     enums are not managed;
         ///     non-enum, non-struct named types are managed;
-        ///     generic types and their nested types are managed;
-        ///     type parameters are managed;
+        ///     type parameters are managed unless an 'unmanaged' constraint is present;
```

Comment:
Some additional comment might be warranted. I think the language you suggest conflates type parameters with type arguments though.

Consider the following example
```cs
struct MyStruct<T>
{
    T field;
    unsafe void Foo(MyStruct<T>* ptr) {}
}

static class MyClass
{
    static unsafe void Bar(MyStruct<int>* ptr) {}
}
```

`MyStruct.Foo` is invalid because the type parameter `T` must have an `unmanaged` constraint to be considered unmanaged. `MyClass.Bar` is fine, though, because the type instantiation `MyStruct<int>` can be statically determined to be unmanaged.

Reply:
Right, that's basically what I meant, I just didn't word it as well as you 😄